### ISSUE TITLE
perlvar - @INC mention of '.'

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#17000]
Migrated from [rt.perl.org#134108](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=134108) (status was 'resolved')

Searchable as RT134108$

**Comment by p5pRT** [comment]
### From @grinnz

In the perlvar entry for @&#8203;INC\, it is stated that '\.' is included at the end
of @&#8203;INC unless taint checks are enabled or Perl is compiled with
\-Ddefault\_inc\_excludes\_dot\. This also should mention that it defaults to
excluding this in Perl 5\.26 or newer\.

\-Dan

**Comment by p5pRT** [comment]
### From @atoomic

On Wed\, 15 May 2019 07&#8203;:21&#8203;:30 \-0700\, grinnz@&#8203;gmail\.com wrote&#8203;:
> In the perlvar entry for @&#8203;INC\, it is stated that '\.' is included at the end
> of @&#8203;INC unless taint checks are enabled or Perl is compiled with
> \-Ddefault\_inc\_excludes\_dot\. This also should mention that it defaults to
> excluding this in Perl 5\.26 or newer\.
> 
> \-Dan

suggested patch

commit 7d55b5be88fe44a69a958b3ea47676df4c64804d \(HEAD \-> perlvar\-inc\)
Author&#8203;: Nicolas R \<atoomic@&#8203;cpan\.org>
Date&#8203;:   Wed May 15 10&#8203;:02&#8203;:55 2019 \-0600

&nbsp;    Adjust perlvar doc for @&#8203;INC

&nbsp;    RT \#134108

&nbsp;    dot is not in @&#8203;INC anymore since 5\.26

<details><summary>Inline Patch</summary>

```diff
diff --git a/pod/perlvar.pod b/pod/perlvar.pod
index 1f9c08ce36..5cc8fabe18 100644
--- a/pod/perlvar.pod
+++ b/pod/perlvar.pod
@@ -494,12 +494,13 @@ The array C<@INC> contains the list of places that the C<do EXPR>,
 C<require>, or C<use> constructs look for their library files.  It
 initially consists of the arguments to any B<-I> command-line
 switches, followed by the default Perl library, probably
-F</usr/local/lib/perl>, followed by ".", to represent the current
-directory.  ("." will not be appended if taint checks are enabled,
-either by C<-T> or by C<-t>, or if configured not to do so by the
-C<-Ddefault_inc_excludes_dot> compile time option.)  If you need to
-modify this at runtime, you should use the C<use lib> pragma to get
-the machine-dependent library properly loaded also:
+F</usr/local/lib/perl>.
+Since Perl 5.26 '.' which represents the current directory
+is not appended anymore to C<@INC>. This is not recommended to change
+this behavior and you can read more about it from
+L<C<PERL_USE_UNSAFE_INC>|perlrun/PERL_USE_UNSAFE_INC>
+If you need to modify C<@INC> at runtime, you should use
+the C<use lib> pragma to get the machine-dependent library properly loaded also:

     use lib '/mypath/libdir/';
     use SomeMod;
```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @atoomic

After some review this is now merged as 87e54f02d

https://github.com/Perl/perl5/commit/87e54f02df2275a05c923ec51e27d3e176499a4b

On Wed\, 15 May 2019 09&#8203;:04&#8203;:02 \-0700\, atoomic wrote&#8203;:
> On Wed\, 15 May 2019 07&#8203;:21&#8203;:30 \-0700\, grinnz@&#8203;gmail\.com wrote&#8203;:
> > In the perlvar entry for @&#8203;INC\, it is stated that '\.' is included at
> > the end
> > of @&#8203;INC unless taint checks are enabled or Perl is compiled with
> > \-Ddefault\_inc\_excludes\_dot\. This also should mention that it defaults
> > to
> > excluding this in Perl 5\.26 or newer\.
> >
> > \-Dan
> 
> suggested patch
> 
> commit 7d55b5be88fe44a69a958b3ea47676df4c64804d \(HEAD \-> perlvar\-inc\)
> Author&#8203;: Nicolas R \<atoomic@&#8203;cpan\.org>
> Date&#8203;:   Wed May 15 10&#8203;:02&#8203;:55 2019 \-0600
> 
> Adjust perlvar doc for @&#8203;INC
> 
> RT \#134108
> 
> dot is not in @&#8203;INC anymore since 5\.26
> 
> diff \-\-git a/pod/perlvar\.pod b/pod/perlvar\.pod
> index 1f9c08ce36\.\.5cc8fabe18 100644
> \-\-\- a/pod/perlvar\.pod
> \+\+\+ b/pod/perlvar\.pod
> @&#8203;@&#8203; \-494\,12 \+494\,13 @&#8203;@&#8203; The array C\<@&#8203;INC> contains the list of places
> that the C\<do EXPR>\,
>  C\<require>\, or C\<use> constructs look for their library files\.  It
>  initially consists of the arguments to any B\<\-I> command\-line
>  switches\, followed by the default Perl library\, probably
> \-F\</usr/local/lib/perl>\, followed by "\."\, to represent the current
> \-directory\.  \("\." will not be appended if taint checks are enabled\,
> \-either by C\<\-T> or by C\<\-t>\, or if configured not to do so by the
> \-C\<\-Ddefault\_inc\_excludes\_dot> compile time option\.\)  If you need to
> \-modify this at runtime\, you should use the C\<use lib> pragma to get
> \-the machine\-dependent library properly loaded also&#8203;:
> \+F\</usr/local/lib/perl>\.
> \+Since Perl 5\.26 '\.' which represents the current directory
> \+is not appended anymore to C\<@&#8203;INC>\. This is not recommended to change
> \+this behavior and you can read more about it from
> \+L\<C\<PERL\_USE\_UNSAFE\_INC>|perlrun/PERL\_USE\_UNSAFE\_INC>
> \+If you need to modify C\<@&#8203;INC> at runtime\, you should use
> \+the C\<use lib> pragma to get the machine\-dependent library properly
> loaded also&#8203;:
> 
> use lib '/mypath/libdir/';
> use SomeMod;

**Comment by p5pRT** [comment]
### From @atoomic

we can close this case \[do not have permissions to do it\]
 
On Thu\, 16 May 2019 10&#8203;:27&#8203;:08 \-0700\, atoomic wrote&#8203;:
> After some review this is now merged as 87e54f02d
> 
> https://github.com/Perl/perl5/commit/87e54f02df2275a05c923ec51e27d3e176499a4b
> 
> On Wed\, 15 May 2019 09&#8203;:04&#8203;:02 \-0700\, atoomic wrote&#8203;:
> > On Wed\, 15 May 2019 07&#8203;:21&#8203;:30 \-0700\, grinnz@&#8203;gmail\.com wrote&#8203;:
> > > In the perlvar entry for @&#8203;INC\, it is stated that '\.' is included at
> > > the end
> > > of @&#8203;INC unless taint checks are enabled or Perl is compiled with
> > > \-Ddefault\_inc\_excludes\_dot\. This also should mention that it
> > > defaults
> > > to
> > > excluding this in Perl 5\.26 or newer\.
> > >
> > > \-Dan
> >
> > suggested patch
> >
> > commit 7d55b5be88fe44a69a958b3ea47676df4c64804d \(HEAD \-> perlvar\-inc\)
> > Author&#8203;: Nicolas R \<atoomic@&#8203;cpan\.org>
> > Date&#8203;:   Wed May 15 10&#8203;:02&#8203;:55 2019 \-0600
> >
> > Adjust perlvar doc for @&#8203;INC
> >
> > RT \#134108
> >
> > dot is not in @&#8203;INC anymore since 5\.26
> >
> > diff \-\-git a/pod/perlvar\.pod b/pod/perlvar\.pod
> > index 1f9c08ce36\.\.5cc8fabe18 100644
> > \-\-\- a/pod/perlvar\.pod
> > \+\+\+ b/pod/perlvar\.pod
> > @&#8203;@&#8203; \-494\,12 \+494\,13 @&#8203;@&#8203; The array C\<@&#8203;INC> contains the list of places
> > that the C\<do EXPR>\,
> >  C\<require>\, or C\<use> constructs look for their library files\.  It
> >  initially consists of the arguments to any B\<\-I> command\-line
> >  switches\, followed by the default Perl library\, probably
> > \-F\</usr/local/lib/perl>\, followed by "\."\, to represent the current
> > \-directory\.  \("\." will not be appended if taint checks are enabled\,
> > \-either by C\<\-T> or by C\<\-t>\, or if configured not to do so by the
> > \-C\<\-Ddefault\_inc\_excludes\_dot> compile time option\.\)  If you need to
> > \-modify this at runtime\, you should use the C\<use lib> pragma to get
> > \-the machine\-dependent library properly loaded also&#8203;:
> > \+F\</usr/local/lib/perl>\.
> > \+Since Perl 5\.26 '\.' which represents the current directory
> > \+is not appended anymore to C\<@&#8203;INC>\. This is not recommended to
> > change
> > \+this behavior and you can read more about it from
> > \+L\<C\<PERL\_USE\_UNSAFE\_INC>|perlrun/PERL\_USE\_UNSAFE\_INC>
> > \+If you need to modify C\<@&#8203;INC> at runtime\, you should use
> > \+the C\<use lib> pragma to get the machine\-dependent library properly
> > loaded also&#8203;:
> >
> > use lib '/mypath/libdir/';
> > use SomeMod;

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
